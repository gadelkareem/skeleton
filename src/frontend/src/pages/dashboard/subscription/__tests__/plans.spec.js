import { mount } from '@vue/test-utils'
import initData from '@@/globals/init-data'
import Vuex from 'vuex'
import VueRouter from 'vue-router'
import Vuetify from 'vuetify'
import Page from '../plans'

const vuetify = new Vuetify()

describe('plans.vue', () => {
  let store

  beforeEach(() => {
    const actions = {
      'page/title': jest.fn(),
      'page/hideFooter': jest.fn(),
      'loading/start': jest.fn(),
      'loading/finish': jest.fn()
    }
    const getters = {
      'user/user': () => initData().user
    }
    store = new Vuex.Store({
      actions,
      getters,
      state: {
        loading: { status: false }
      }
    })
  })

  it('should match snapshot', () => {
    const router = new VueRouter({})
    const w = mount(Page, {
      store,
      router,
      vuetify,
      stubs: ['router-link', 'router-view'],
      created () {
        this.$vuetify.lang = {
          t: () => {}
        }
        this.$vuetify.breakpoint = {
          width: 0,
          height: 0
        }
        this.$vuetify.theme = { dark: false }
      }
    })

    expect(w.html()).toMatchSnapshot()
    w.destroy()
  })
})
