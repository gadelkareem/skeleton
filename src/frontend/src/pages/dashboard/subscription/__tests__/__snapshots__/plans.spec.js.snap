// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`plans.vue should match snapshot 1`] = `
<section class="container container--fluid" id="create-subscription"><span> <!----> <!----></span> <span style="display: none;"><h3><span class="step__number"></span>
  Select A Plan
  </h3>
  <div class="v-item-group theme--light">
    <div class="container">
      <div class="row"></div>
    </div>
  </div>
  <!---->
  </span>
  <!---->
  <!---->
  <div class="spacer" style="margin-top: 200px;"></div>
  <!---->
</section>
`;
